test/integ/test_cursor.py

```diff
@@ -1206,6 +1206,25 @@ def test_check_can_use_pandas(conn_cnx):
                 cur.check_can_use_pandas()
 
 
+@pytest.mark.skipolddriver
+def test_execute_async_and_fetch_pandas_batches(conn_cnx):
```

Comment:
Tests are failing because you need to move this test inside of `pandas` sub-directory. I'd say it should be in https://github.com/snowflakedb/snowflake-connector-python/blob/main/test/integ/pandas/test_arrow_pandas.py